azurerm/resource_arm_servicebus_rule_test.go

```diff
@@ -0,0 +1,322 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMServiceBusTopicDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMServiceBusRuleExists(),
+					resource.TestCheckResourceAttr(resourceName, "correlation_filter.0.message_id", "test_message_id"),
+					resource.TestCheckResourceAttr(resourceName, "correlation_filter.0.reply_to", ""),
+				),
+			},
+			{
+				Config: updatedConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMServiceBusRuleExists(),
+					resource.TestCheckResourceAttr(resourceName, "correlation_filter.0.message_id", "test_message_id_updated"),
+					resource.TestCheckResourceAttr(resourceName, "correlation_filter.0.reply_to", "test_reply_to_added"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMServiceBusRule_updateSqlFilterToCorrelationFilter(t *testing.T) {
+	ri := acctest.RandInt()
+	location := testLocation()
+	config := testAccAzureRMServiceBusRule_basicSqlFilter(ri, location)
+	updatedConfig := testAccAzureRMServiceBusRule_basicCorrelationFilter(ri, location)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMServiceBusTopicDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMServiceBusRuleExists(),
+				),
+			},
+			{
+				Config: updatedConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMServiceBusRuleExists(),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMServiceBusRuleExists() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		ruleName := rs.Primary.Attributes["name"]
+		subscriptionName := rs.Primary.Attributes["subscription_name"]
+		topicName := rs.Primary.Attributes["topic_name"]
+		namespaceName := rs.Primary.Attributes["namespace_name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Rule: %q", ruleName)
+		}
+
+		client := testAccProvider.Meta().(*ArmClient).serviceBusRulesClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := client.Get(ctx, resourceGroup, namespaceName, topicName, subscriptionName, ruleName)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on serviceBusRulesClient: %+v", err)
+		}
+
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Bad: Rule %q (resource group: %q) does not exist", ruleName, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMServiceBusRule_basicSqlFilter(rInt int, location string) string {
+	sqlFilter := `
+	filter_type = "SqlFilter"
+	sql_filter  = "2=2"
+	`
```

Comment:
rather than templating this - can we update this to be the full test in-line? for instance

```
resource "azurerm_resource_group" "test" {
  # ..
}

resource "azurerm_servicebus_subscription_rule" "test" {
  # ..
}
```